packages/react-button/src/components/Button/ButtonBase.tsx

```diff
@@ -6,22 +6,22 @@ import { useButton } from './useButton';
 export const ButtonBase = compose<'button', ButtonProps, ButtonProps, {}, {}>(
   (props, ref, options) => {
     const { state } = options;
+    const { children, icon, iconOnly, iconPosition, loading } = state;
     const { slots, slotProps } = mergeProps<ButtonProps, ButtonProps, ButtonSlots, ButtonSlotProps>(state, options);
 
     return (
       <slots.root ref={ref} {...slotProps.root}>
-        {props.loading && <slots.loader {...slotProps.loader} />}
-        {props.icon && props.iconPosition !== 'after' && <slots.icon {...slotProps.icon} />}
-        {!props.iconOnly && props.content && <slots.content {...slotProps.content} />}
-        {props.icon && props.iconPosition === 'after' && <slots.icon {...slotProps.icon} />}
+        {loading && <slots.loader {...slotProps.loader} />}
+        {icon && iconPosition !== 'after' && <slots.icon {...slotProps.icon} />}
+        {!iconOnly && children && <span>{children}</span>}
+        {icon && iconPosition === 'after' && <slots.icon {...slotProps.icon} />}
       </slots.root>
     );
   },
   {
     displayName: 'ButtonBase',
     handledProps: [
       'circular',
-      'content',
```

Comment:
I don't understand why we are `removing` the content, just because we want to support `children`. In my opinion, if there are children, we should just render children (that way we are allowing the users to opt out of our rendering logic). If they want to use shorthand, they can use the `content` in combination with other slots.

Reply:
The reasoning we had was that we shouldn't go against the standard expectations. When talking about an `HTML` button we expect the content to be passed as `children`. Having both a `content` slot and `children` rendering support seems unintuitive and would most probably lead to confusion, especially for first-time users of our components.

An example: If we just let them opt out of our rendering logic what's the expectation? Should we still render the `icons` or not? Where's the position of the `children` if the answer to the previous question is yes? It all can become very confusing very fast.

Reply:
I think replacing `content` with only `children` is the right call - it simplifies the API and is more predictable.

If you try other component library buttons, or just vanilla, Rendering a button with some text is always this:

```tsx
<Button>Hello</Button>
```

Take a look at other frameworks as well; this seems to be a consistent pattern. 

A user will then add an icon:

```tsx
<Button>
  <AddIcon />
  Hello
</Button>
```

This creates a wrong looking button. The icon size isn't contained, may not be the right color, doesn't have the right gap. So they realize we have a slot for that:

```tsx
<Button icon={ <AddIcon/> }>Hello</Button>
```

This feels ergonomic and matches other libraries. So, I believe having both content and children definitely creates redundancy, but just having children seems like a pattern more consistent with native.

There are cases like `Checkbox`, where `children` makes less sense. I would expect cases like this prohibiting `children` because it tends to follow the native `input` element expectations.

`Link` is another analogous example which wouldn't make sense to prioritize `content` over `children`. I think `Button` and `Link` should feel pretty symmetrical.